### ISSUE TITLE
Filter for the side menu

### ISSUE THREAD

**Issue by benjore** [unfoldingWord/translationCore#1998]
## Breakdown
- [x] M Badges for the icons in the menu with hover-over
- [x] L Expanded Checkbox filter
- [x] M Collapsed menu with "Chips"

## Specs
- [x] When multiple filters are active, they are OR (not AND), e.g. display all checks with a bookmark OR a comment
- [x] For all checks with more than one icon, show hidden icons with a number and a pop-over showing the "hidden" icons (see last image)
- [x] NOTE: If "Selected" is checked, then "No Selection" must become deactivated. Conversely, if "No Selection" is checked, then "Selected" must become deactivated. 

![image](https://user-images.githubusercontent.com/13352518/34792736-9deb1d20-f60e-11e7-91d0-cbe1fcfc303c.png)

![image](https://user-images.githubusercontent.com/13352518/34792743-a6cb2034-f60e-11e7-8e9d-a1653a6d6c67.png)

![image](https://user-images.githubusercontent.com/13352518/34792810-d93597ca-f60e-11e7-95b9-6166c02c7f6a.png)

![image](https://user-images.githubusercontent.com/13352518/34792824-e15ab160-f60e-11e7-8993-b35655bda8bd.png)

![image](https://user-images.githubusercontent.com/13352518/35296225-04559126-0041-11e8-9942-e5667a8eba3f.png)

![image](https://user-images.githubusercontent.com/13352518/34792862-115d585e-f60f-11e7-8395-c9c49abdb701.png)

![image](https://user-images.githubusercontent.com/13352518/34792869-1944847a-f60f-11e7-8b3b-5199f9e443ce.png)

![image](https://user-images.githubusercontent.com/13352518/34792879-200e462e-f60f-11e7-90e8-5ac04b1b2cf8.png)

![image](https://user-images.githubusercontent.com/13352518/34792895-271624f0-f60f-11e7-8023-ef89eee2d28d.png)

![image](https://user-images.githubusercontent.com/13352518/35295902-1bb4768a-0040-11e8-9ad4-eff231d7187f.png)

![image](https://user-images.githubusercontent.com/13352518/35296091-93cc5a02-0040-11e8-9d4a-cee379353c06.png)

**Comment by richmahn** [comment]
@benjore  Is adding a data file for all invalid verses of a project in the scope of epic #1998? And when does a verse become "valid" again (that is, the broken link icon disappears). If text changed, we wouldn't know what needs to be highlighted to make it valid, would we? Or just that something gets highlighted? This adds another issue to this epic if we need to provide invalid verse link icons in tW.

**Comment by benjore** [comment]
@richmahn That is only in scope for #3086 . This issue assumes that issue is complete (though I don't think it is blocked by it not being complete).

**Comment by richmahn** [comment]
@benjore Ok, will try to include what I can in this story to assume there is info on invalidity.

**Comment by richmahn** [comment]
@benjore We still need to figure out what to do with the invalidated status which you have showing a unlink icon, but Glyph icons for bootstrap don't have any unlink, just a link: https://getbootstrap.com/docs/3.3/components/

Currently I just put "flag" in as the icon for when #3086 gets done.

**Comment by richmahn** [comment]
Just to also put it here, the order of the icons when there are multiple (e.g. which one is the main glyph if there are more than one status badges):

```
That order has already been set (minus Invalidated which is new). But to make it explicit. 
1. Invalidated
2. Bookmark
3. Selected
4. Verse Edit
5. Comment
```

**Comment by richmahn** [comment]
@benjore I don't see a reason to do this whole if showing selected then don't show not selected. How does the app even start in the tool if only one or the other if shown? Don't show selected? I have been using it where you can use both or either and it works as you'd expect. Having then both enabled is like 0.9 shows everything. Doesn't make sense to show one or the other. It's true that if both selected then the other filters are mute, but the user would just have to know they need to infect one or the other or both. The other way I guess might be possible is if NO filters are selected then we show everything and start the tool with them all off?

Speaking of starting the tool, should the filter part of the menu be expanded or collapsed when it opens? I prefer collapsed myself.

**Comment by cckozie** [comment]
All child issues complete and passed.
